docs/user/site_config.rst

```diff
@@ -116,6 +115,11 @@ The `hub` section defines the connection parameters for the crossbar server.
 This entire section will likely remain unchanged, except for the
 ``wamp_server`` and ``wamp_http`` IP addresses.
 
+The `address_root` setting determines the leading token in all agent
+and feed addresses on the crossbar network.  While "observatory" is
+the default, it can be changed as long as the crossbar configuration
+is also updated to permit operations on the `{address_root}.` uri.
```

Comment:
I suppose my only concern now is users now having to configure crossbar, which we were avoiding previously. I think it would be easy to miss this requirement that the crossbar URI needs to change, especially since users haven't been dealing with crossbar configs previously.

`ocsbow status` will detect a potential mismatch, but just looking at agent log output it looks like a simple crossbar server not available and attempted reconnect:
```
2023-05-12T21:12:08+0000 session joined: {'authextra': {'x_cb_node': 'efc3505b29a0-1',
               'x_cb_peer': 'tcp4:192.168.96.7:57996',
               'x_cb_pid': 17,
               'x_cb_worker': 'worker001'},
 'authid': 'HG7R-VKER-KUTG-FR4V-JHS7-M7RP',
 'authmethod': 'anonymous',
 'authprovider': 'static',
 'authrole': 'iocs_agent',
 'realm': 'test_realm',
 'resumable': False,
 'resume_token': None,
 'resumed': False,
 'serializer': 'msgpack.batched',
 'session': 7293750874092658,
 'transport': {'channel_framing': 'websocket',
               'channel_id': {},
               'channel_serializer': None,
               'channel_type': 'tcp',
               'http_cbtid': None,
               'http_headers_received': None,
               'http_headers_sent': None,
               'is_secure': False,
               'is_server': False,
               'own': None,
               'own_fd': -1,
               'own_pid': 7,
               'own_tid': 7,
               'peer': 'tcp4:192.168.96.2:8001',
               'peer_cert': None,
               'websocket_extensions_in_use': None,
               'websocket_protocol': None}}
2023-05-12T21:12:08+0000 Failed to register basic handlers @ test2.fake-data1; agent probably running already.
2023-05-12T21:12:08+0000 session left: CloseDetails(reason=<wamp.close.normal>, message='None')
2023-05-12T21:12:08+0000 transport disconnected
2023-05-12T21:12:08+0000 waiting for reconnect for 9.999993801116943 more seconds
2023-05-12T21:12:08+0000 Scheduling retry 1 to connect <twisted.internet.endpoints.TCP4ClientEndpoint object at 0x7fc9ac4b2f10> in 2.247658769532999 seconds.
2023-05-12T21:12:08+0000 waiting for reconnect for 1.9931435585021973 more seconds
2023-05-12T21:12:08+0000 waiting for reconnect for 4.9938740730285645 more seconds
2023-05-12T21:12:08+0000 waiting for reconnect for 2.9927923679351807 more seconds
2023-05-12T21:12:09+0000 waiting for reconnect for 6.997904062271118 more seconds
2023-05-12T21:12:09+0000 waiting for reconnect for 8.998390197753906 more seconds
2023-05-12T21:12:09+0000 waiting for reconnect for 0.9915776252746582 more seconds
2023-05-12T21:12:09+0000 waiting for reconnect for 3.992349624633789 more seconds
2023-05-12T21:12:09+0000 waiting for reconnect for 1.9910469055175781 more seconds
2023-05-12T21:12:10+0000 waiting for reconnect for 5.996617317199707 more seconds
2023-05-12T21:12:10+0000 waiting for reconnect for 7.997318506240845 more seconds
2023-05-12T21:12:10+0000 stopping reactor
2023-05-12T21:12:10+0000 Main loop terminated.
```

Any ideas on how to improve this on the user end? Can we improve the error message there? And perhaps revisit the documentation on crossbar configuration, `ocsbow crossbar generate_config` is still documented there and seems like its deprecated:
```
$ ocsbow crossbar generate_config
usage: ocsbow [-h] [--site SITE] [--site-file SITE_FILE] [--site-host SITE_HOST] [--site-hub SITE_HUB] [--site-http SITE_HTTP] [--site-realm SITE_REALM]
              [--instance-id INSTANCE_ID] [--address-root ADDRESS_ROOT] [--registry-address REGISTRY_ADDRESS] [--log-dir LOG_DIR] [--working-dir WORKING_DIR]
              {status,up,down,config} ...
ocsbow: error: argument command: invalid choice: 'crossbar' (choose from 'status', 'up', 'down', 'config')
```

And the newer `ocs-local-support generate_crossbar_config` seems to not work either:
```
$ ocs-local-support generate_crossbar_config
Traceback (most recent call last):
  File "/home/koopman/git/ocs/.venv/bin/ocs-local-support", line 8, in <module>
    sys.exit(main_local())
  File "/home/koopman/git/ocs/.venv/lib/python3.10/site-packages/ocs/ocsbow.py", line 1014, in main_local
    cm = supports.crossbar['ctrl']
KeyError: 'ctrl'
```

Reply:
Maybe we can build an entrypoint script for the crossbar container that allows for setting the address root in the crossbar container's environment variables? Then users would have to have matching address roots in the SCF and the docker-compose file for their crossbar server, which would look like:
```
  crossbar:
    image: simonsobs/ocs-crossbar:latest
    ports:
      - "127.0.0.1:8001:8001"
    environment:
         - PYTHONUNBUFFERED=1
         - ADDRESS_ROOT=lab1
```

Reply:
Ok, see what I've done in latest commit.  Happy to change envvar names, etc.

I updated the docs, minimally, but I think:
- We should drop the part about rebuilding images with modified config.json ... an advanced docker user would already know how to do that, and bind-mounting is good enough for most OCS users.
- We should move the docker part to be before the generate_crossbar_config stuff.

Reply:
Looks great! Thanks for the updates.

I agree on both those points. The link in the "Example Config" section is going to break now too. Since `ocs/docker/crossbar/config.json` and `ocs/ocs/support/crossbar_config.json` are the same now, I'd change some of the wording in that section, and maybe just "include" the file on that page for viewing, rather than link to it.

Do you want to make those docs changes, or shall I? Happy to help there.

Reply:
> Do you want to make those docs changes, or shall I? Happy to help there.

Cool!  If you could take over from here I think that would be the most efficient way forward.